### PR TITLE
Use chainable proxy in `appRouter.createCaller()`

### PR DESCRIPTION
**Pull request by KATT**

### CODE REVIEW

File: examples/.interop/next-prisma-starter/src/server/routers/post.test.ts [hunk 1]
```diff
@@ -13,10 +13,8 @@ test('add and get post', async () => {
     text: 'hello test',
     title: 'hello test',
   };
-  const post = await caller.mutation('post.add', input);
-  const byId = await caller.query('post.byId', {
-    id: post.id,
-  });
+  const post = await caller['post.add'](input);
+  const byId = await caller['post.byId']({ id: post.id });
```
Comment (KATT) [inline examples/.interop/next-prisma-starter/src/server/routers/post.test.ts:17]: cc @sachinraja 


Shouldn't this be?:

```suggestion
  const post = await caller.post.add(input);
  const byId = await caller.post.byId({ id: post.id });
```

❓

Comment (KATT) [reply examples/.interop/next-prisma-starter/src/server/routers/post.test.ts:17]: Related: #2284

cc @IgnusG

Comment (sachinraja) [reply examples/.interop/next-prisma-starter/src/server/routers/post.test.ts:17]: Yes but I'm not sure it's worth it to implement. Type perf is already worsened on the interop routers. We'd have to replicate the structure of a nested router, derived from an interop router.

Comment (KATT) [reply examples/.interop/next-prisma-starter/src/server/routers/post.test.ts:17]: Alright, let's take this as a task on the roadmap

**Pull request closed without merging**